pyart/graph/radarmapdisplay_cartopy.py

```diff
@@ -0,0 +1,510 @@
+            standard_parallels=(lat_0, ))
+        self.proj = proj
+
+        # create the basemap if not provided
+        import matplotlib.pyplot as plt
+
+        if hasattr(ax, 'projection'):
+            basemap = ax
+        elif hasattr(ax, 'projection'):
+            basemap = basemap
+        else:
+            if projection is None:
+                basemap = plt.axes(projection=proj)
+            else:
+                basemap = plt.axes(projection=projection)
+        self.basemap = basemap
+        if min_lon:
+            basemap.set_extent([min_lon, max_lon, min_lat, max_lat],
+                               crs=cartopy.crs.PlateCarree())
+
+        # add embelishments
+        if embelish is True:
+            # shpfile = cartopy.io.shapereader.gshhs(resolution)
+            # shp = cartopy.io.shapereader.Reader(shpfile)
+            # basemap.add_geometries(
+            #    shp.geometries(), cartopy.crs.PlateCarree(), edgecolor='black',
+            #    facecolor='none')
+            #basemap.drawstates()
+            if basemap.projection in [cartopy.crs.PlateCarree(),
+                                      cartopy.crs.Mercator()]:
+                gl = basemap.gridlines(xlocs=lon_lines, ylocs=lat_lines,
+                                       draw_labels=True)
+                gl.xlabels_top = False
+                gl.ylabels_right = False
+
+            elif basemap.projection == proj:
+                fig.canvas.draw()
+                basemap.gridlines(xlocs=lon_lines, ylocs=lat_lines)
+
+                # Label the end-points of the gridlines using the custom
+                # tick makers:
+                basemap.xaxis.set_major_formatter(
+                    cartopy.mpl.gridliner.LONGITUDE_FORMATTER)
+                basemap.yaxis.set_major_formatter(
+                    cartopy.mpl.gridliner.LATITUDE_FORMATTER)
+                lambert_xticks(basemap, lon_lines)
+                lambert_yticks(basemap, lat_lines)
+
+            else:
+                basemap.gridlines(xlocs=lon_lines, ylocs=lat_lines)
+
+            # Create a feature for States/Admin 1 regions at 1:resolution
+            # from Natural Earth
+            states_provinces = cartopy.feature.NaturalEarthFeature(
+                category='cultural',
+                name='admin_1_states_provinces_lines',
+                scale=resolution,
+                facecolor='none')
+            basemap.coastlines(resolution=resolution)
+            basemap.add_feature(states_provinces, edgecolor='gray')
+
+        # plot the data and optionally the shape file
+        # we need to convert the radar gate locations (x and y) which are in
+        # km to meters as well as add the map coordiate radar location
+        # which is given by self._x0, self._y0.
+        pm = basemap.pcolormesh(
+            x * 1000., y * 1000., data,
+            vmin=vmin, vmax=vmax, cmap=cmap, norm=norm, transform=proj)
```

Comment:
`transform` here should be the original projection of the data, which is always Lambert Conformal right? The map will be drawn in the projection of the `self.basemap` object, so we just need to know what projection we are coming from